misc/testerina/modules/testerina-core/src/main/ballerina/assert.bal

```diff
@@ -309,48 +310,48 @@ isolated function getMapValueDiff(map<anydata> actualMap, map<anydata> expectedM
     string[] expectedKeyArray = getKeyArray(expectedMap);
     string keyDiff = getKeysDiff(actualKeyArray, expectedKeyArray);
     string valueDiff = compareMapValues(actualMap, expectedMap);
-    if (keyDiff != "") {
+    if keyDiff != "" {
         diffValue = diffValue.concat(keyDiff, "\n", valueDiff);
     } else {
         diffValue = diffValue.concat(valueDiff);
     }
     return diffValue;
 }
 
-isolated function getValueComparison(anydata actual, anydata expected, string keyVal, int count) returns @tainted ([string, int])  {
+isolated function getValueComparison(anydata actual, anydata expected, string keyVal, int count) 
+        returns @tainted ([string, int]) {
     int diffCount = count;
     string diff = "";
     string expectedType = getBallerinaType(expected);
     string actualType = getBallerinaType(actual);
-    if (expectedType != actualType) {
+    if expectedType != actualType {
         diff = diff.concat("\n", "key: ", keyVal, "\n \nexpected value\t: <", expectedType, "> ", expected.toString(),
         "\nactual value\t: <", actualType, "> ", actual.toString());
         diffCount = diffCount + 1;
-    } else {
-        if (actual is map<anydata> && expected is map<anydata>) {
-            string[] expectedkeyArray = (<map<anydata>>expected).keys();
-            string[] actualKeyArray = (<map<anydata>>actual).keys();
-            int orderCount = diffCount;
-            foreach string childKey in actualKeyArray {
-                if (expectedkeyArray.indexOf(childKey) != ()){
-                    anydata expectedChildVal = expected.get(childKey);
-                    anydata actualChildVal = actual.get(childKey);
-                    string childDiff;
-                    if (expectedChildVal != actualChildVal) {
-                        [childDiff, diffCount] = getValueComparison(actualChildVal, expectedChildVal, keyVal + "." + childKey, diffCount);
-                        if (diffCount != (orderCount + 1)) {
-                            diff = diff.concat("\n");
-                        }
-                        diff = diff.concat(childDiff);
+    } else if actual is map<anydata> && expected is map<anydata> {
+        string[] expectedkeyArray = (<map<anydata>>expected).keys();
+        string[] actualKeyArray = (<map<anydata>>actual).keys();
```

Comment:
```suggestion
        string[] expectedkeyArray = expected.keys();
        string[] actualKeyArray = actual.keys();
```
The type narrowing should work here.

Reply:
Changed with 27c521a5975fb3e867c8bdc46b9cc676e14f992c